components/notification.php

```diff
@@ -0,0 +1,350 @@
+	 *     Success: Successfully created new notification (ID #48949)
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$r = wp_parse_args( $assoc_args, array(
+			'component'         => '',
+			'action'            => '',
+			'user-id'           => 0,
+			'item-id'           => 0,
+			'secondary-item-id' => 0,
+			'date'              => bp_core_current_time(),
+		) );
+
+		// Fill in the component.
+		if ( empty( $r['component'] ) ) {
+			$r['component'] = $this->get_random_component();
+		}
+
+		$id = bp_notifications_add_notification( array(
+			'user_id'           => $r['user-id'],
+			'item_id'           => $r['item-id'],
+			'secondary_item_id' => $r['secondary-item-id'],
+			'component_name'    => $r['component'],
+			'component_action'  => $r['action'],
+			'date_notified'     => $r['date'],
+		) );
+
+		// Silent it before it errors.
+		if ( WP_CLI\Utils\get_flag_value( $assoc_args, 'silent' ) ) {
+			return;
+		}
+
+		if ( ! is_numeric( $id ) ) {
+			WP_CLI::error( 'Could not create notification.' );
+		}
+
+		if ( WP_CLI\Utils\get_flag_value( $assoc_args, 'porcelain' ) ) {
+			WP_CLI::line( $id );
+		} else {
+			WP_CLI::success( sprintf( 'Successfully created new notification (ID #%d)', $id ) );
+		}
+	}
+
+	/**
+	 * Get specific notification.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <notification-id>
+	 * : Identifier for the notification.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 *  ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - ids
+	 *   - json
+	 *   - count
+	 *   - yaml
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp notification get 500
+	 *     $ wp bp notification get 56 --format=json
+	 *
+	 * @alias see
+	 */
+	public function get( $args, $assoc_args ) {
+		$notification = bp_notifications_get_notification( $args[0] );
+
+		if ( empty( $notification->id ) ) {
+			WP_CLI::error( 'No notification found by that ID.' );
+		}
+
+		if ( ! is_object( $notification ) ) {
+			WP_CLI::error( 'Could not find the notification.' );
+		}
+
+		$notification_arr = get_object_vars( $notification );
+
+		if ( empty( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = array_keys( $notification_arr );
+		}
+
+		$formatter = $this->get_formatter( $assoc_args );
+		$formatter->display_item( $notification_arr );
+	}
+
+	/**
+	 * Delete a notification.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <notification-id>...
+	 * : ID or IDs of notification to delete.
+	 *
+	 * [--yes]
+	 * : Answer yes to the confirmation message.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp notification delete 520
+	 *     Success: Notification deleted.
+	 *
+	 *     $ wp bp notification delete 55654 54564 --yes
+	 *     Success: Notification deleted.
+	 *
+	 * @alias trash
+	 */
+	public function delete( $args, $assoc_args ) {
+		$notification_id = $args[0];
```

Comment:
This doesn't work to allow multiple deletions. I think you need to pass all of `$args` to `parent::_delete()`. This could use a Behat test to confirm.

Reply:
That's true! I'll need to update in the other commands as well.